File: CivitasRuby/lib/sorpresa_carcel.rb
# encoding:utf-8
require_relative 'sorpresa.rb'
module Civitas
  class SorpresaCarcel < Sorpresa
    def initialize(tablero)
      texto = "¡Directo a la cárcel!"
      super(texto)
      @tablero = tablero
    end
    
    def informe(actual, todos)
      Diario.instance.ocurre_evento("Aplicando sorpresa " + self.to_s + 
        " al jugador " + todos[actual].nombre + " de tipo IrACarcel")
    end
    
    def aplicar_a_jugador(actual, todos)      
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos[actual].encarcelar(@tablero.num_casilla_carcel);
      end
    end
    
    def to_s
      "Sorpresa: { \n Texto: #{@texto}  \n Tipo: IrACarcel}"
    end
  end
end

File: CivitasJava/src/Civitas/SorpresaCarcel.java
package Civitas;

import java.util.ArrayList;

public class SorpresaCarcel extends Sorpresa{
    private Tablero tablero;
    private String texto = "¡Directo a la cárcel!";
    
    SorpresaCarcel (Tablero tablero){
        super("¡Directo a la cárcel!");
        this.tablero = tablero;
    }
        
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(super.jugadorCorrecto(actual, todos)){
            informe(actual,todos);
            todos.get(actual).encarcelar(this.tablero.getCarcel());
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa: { \n Texto = " + texto + "\n Tipo = IrACarcel }";
    }
}
    
    
